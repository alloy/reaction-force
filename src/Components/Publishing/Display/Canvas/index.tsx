import React from "react"
import styled, { StyledFunction } from "styled-components"
import Colors from "../../../../Assets/Colors"
import { pMedia } from "../../../Helpers"
import { Fonts } from "../../Fonts"
import { CanvasContainer } from "./CanvasContainer"

interface DisplayCanvasProps {
  unit: any
  campaign: any
}

export const DisplayCanvas: React.SFC<DisplayCanvasProps> = props => {
  const { unit, campaign } = props
  const disclaimer = <Disclaimer layout={unit.layout}>{unit.disclaimer}</Disclaimer>

  return (
    <DisplayContainer layout={unit.layout}>
      <a href={unit.link.url} target="_blank">
        <AdvertisementBy>{`Advertisement by ${campaign.name}`}</AdvertisementBy>
      </a>
      <CanvasContainer unit={unit} campaign={campaign} disclaimer={disclaimer} />
      {unit.layout === "overlay" && disclaimer}
    </DisplayContainer>
  )
}

interface DivProps extends React.HTMLProps<HTMLDivElement> {
  layout: string
}

const Div: StyledFunction<DivProps> = styled.div

const DisplayContainer = Div`
  display: flex;
  flex-direction: column;
  width: 100%;
  max-width: ${props => (props.layout === "slideshow" ? "100%;" : "1250px;")}
  margin: 0 auto;
  margin-bottom: -44px; // Offset default force margin; FIXME: This kind of stuff needs to be globally addressed
  box-sizing: border-box;
  a {
    text-decoration: none;
  }
  ${props => pMedia.sm`
    ${props.layout !== "slideshow" && "padding: 0 20px;"}
    margin-bottom: 0;
  `}
`
const AdvertisementBy = styled.div`
  ${Fonts.avantgarde("s11")}
  color: ${Colors.grayMedium};
  margin: 10px 0;
  text-align: center;
`
const Disclaimer = Div`
  ${Fonts.garamond("s11")}
  color: ${Colors.grayMedium};
  margin: 15px 0 0 0;
  ${props => props.layout === "overlay" && "text-align: center;"}
  ${pMedia.sm`
    margin: 35px 0 10px 0;
  `}
`