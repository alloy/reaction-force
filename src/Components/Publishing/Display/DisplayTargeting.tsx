import { data as sd } from "sharify"

export const targetingData = (id: string, pageType: string) => {
  const isTesting = JSON.parse(sd.HASHTAG_LAB_ADS_ENABLED || "true")
  return {
    is_testing: isTesting,
    page_type: pageType,
    post_id: id,
  }
}
