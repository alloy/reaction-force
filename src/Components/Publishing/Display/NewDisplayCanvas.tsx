import React, { SFC } from "react"
import { DisplayAd, DisplayAdProps } from "./DisplayAd"

export const NewDisplayCanvas: SFC<DisplayAdProps> = props => (
  <DisplayAd {...props} pt={4} />
)
