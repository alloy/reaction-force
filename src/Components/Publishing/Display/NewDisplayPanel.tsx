import React, { SFC } from "react"
import { DisplayAd, DisplayAdProps } from "./DisplayAd"

export const NewDisplayPanel: SFC<DisplayAdProps> = props => (
  <DisplayAd {...props} />
)
