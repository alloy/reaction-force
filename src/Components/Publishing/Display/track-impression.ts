const alreadyFired = {}

/**
 * An impression tracking utility for display ads that tries to not double 
 * track the same impression by checking a cache of previous impressions.
 * 
 * @example
 *
 *      import { track } from "src/utils/track"
 *
 *      @track()
 *      class DisplayAd extends React.Component<{}, null> {
 * 
 *        @trackImpression(() => "panel")
 *        componentDidMount() {
 *          // ...
 *        }
 *      }
 */
export function trackImpression(unitLayout) {
  return (target, name, descriptor) => {
    const decoratedFn = descriptor.value
    // tslint:disable-next-line:only-arrow-functions
    descriptor.value = function() {
      const key = [
        this.props.campaign.name,
        unitLayout(this.props),
        (this.props.article && this.props.article.id) || this._reactInternalInstance._debugID,
      ].join(":")
      if (alreadyFired[key]) return decoratedFn.apply(this, arguments)
      this.props.tracking &&
        this.props.tracking.trackEvent({
          action: "Impression",
          entity_type: "display_ad",
          campaign_name: this.props.campaign.name,
          unit_layout: unitLayout(this.props),
        })
      alreadyFired[key] = true
      decoratedFn.apply(this, arguments)
    }
  }
}
