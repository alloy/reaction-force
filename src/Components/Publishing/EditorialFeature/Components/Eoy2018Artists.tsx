import {
  ArticleProps,
  FeatureLayout,
} from "Components/Publishing/Layouts/FeatureLayout"
import { Nav } from "Components/Publishing/Nav/Nav"
import React from "react"
import styled from "styled-components"

export const Eoy2018Artists: React.SFC<ArticleProps> = props => {
  return (
    <ArticleWrapper>
      <Nav canFix={false} transparent />
      <FeatureLayout {...props} />
    </ArticleWrapper>
  )
}

const ArticleWrapper = styled.div`
  background: blue;
  color: white;
  padding-top: 75px;

  .content-end {
    background: white;
  }

  div[class^="FeatureInnerContent__"] {
    color: white;
  }
  div[class^="Author__"] {
    color: white;
    &:before {
      background-color: white;
    }
  }
  .Byline {
    color: white;
    svg path {
      fill: white;
    }
  }
`
