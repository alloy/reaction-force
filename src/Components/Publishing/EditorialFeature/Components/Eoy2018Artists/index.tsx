import { color, Flex, Serif, space } from "@artsy/palette"
import { compact, find, map } from "lodash"
import React from "react"
import styled from "styled-components"

import { unica } from "Assets/Fonts"
import { Byline, BylineContainer } from "Components/Publishing/Byline/Byline"
import { ArticleProps } from "Components/Publishing/Layouts/FeatureLayout"
import { Nav } from "Components/Publishing/Nav/Nav"
import { ImageSetPreview } from "Components/Publishing/Sections/ImageSetPreview"
import { resize } from "Utils/resizer"
import { Eoy2018ArticleHeader } from "./ArticleHeader"

export class Eoy2018Artists extends React.Component<ArticleProps> {
  getHeaderSections = () => {
    const { sections } = this.props.article
    const headers = []

    sections.map((section, index) => {
      let headerSection
      const sectionAfter = sections[index + 1]
      const sectionAfterisImage =
        sectionAfter && sectionAfter.type === "image_collection"

      if (this.sectionIsHeader(section)) {
        headerSection = {
          index,
          section,
        }
        if (sectionAfterisImage) {
          headerSection.imageSection = sectionAfter
        }
        headers.push(headerSection)
      }
    })
    return headers
  }

  sectionIsHeader = section => {
    const isText = section.type === "text"
    const isHeader = isText && section.body.includes("<h1>")
    return isHeader
  }

  sectionArtistHeader = (section, i) => {
    const headerSections = this.getHeaderSections()
    const { imageSection } = find(headerSections, ["section", section])
    const src = imageSection && imageSection.images[0].url

    return (
      <ArtistHeaderSection key={i}>
        <ArtistHeaderTitle dangerouslySetInnerHTML={{ __html: section.body }} />
        <ArtistHeaderImg src={src && resize(src, { width: 700 })} />
      </ArtistHeaderSection>
    )
  }

  sectionText = (section, i) => {
    return (
      <TextSection size="5" key={i}>
        <div dangerouslySetInnerHTML={{ __html: section.body }} />
      </TextSection>
    )
  }

  sectionImageSet = (section, i) => {
    return (
      <ImageSetWrapper key={i}>
        <ImageSetPreview section={section} />
      </ImageSetWrapper>
    )
  }

  getSections = () => {
    const {
      article: { sections },
    } = this.props
    const headerSections = this.getHeaderSections()
    const headerTextIndexes = map(headerSections, "index")
    const headerImages = map(headerSections, "image")

    const renderedSections = sections.map((section, i) => {
      const isHeader = headerTextIndexes.includes(i)
      const isHeaderImage = headerImages.includes(section)

      if (isHeader) {
        return this.sectionArtistHeader(section, i)
      } else if (isHeaderImage) {
        return null
      } else {
        if (section.type === "text" && i !== 0) {
          return this.sectionText(section, i)
        } else if (section.type === "image_set") {
          return this.sectionImageSet(section, i)
        } else {
          return null
        }
      }
    })
    return renderedSections
  }

  render() {
    const { article } = this.props
    const introText = this.sectionText(article.sections[0], 0)
    const headerImages = map(
      compact(map(this.getHeaderSections(), "imageSection")),
      "images"
    )

    return (
      <React.Fragment>
        <Nav canFix color={color("black100")} backgroundColor="white" />
        <ArticleWrapper>
          <Eoy2018ArticleHeader images={headerImages} />
          <ArticleContent>
            <IntroSection alignItems="flex-start" pl={20}>
              <Byline article={article} />
              {introText}
            </IntroSection>
            {this.getSections()}
          </ArticleContent>
        </ArticleWrapper>
      </React.Fragment>
    )
  }
}

const BORDER_WIDTH = 6

const ArticleWrapper = styled.div`
  padding: 0 40px;
`

const ArticleContent = styled.div`
  padding: 40px 0;
  border-left: ${BORDER_WIDTH}px solid ${color("purple100")};

  blockquote {
    ${unica("s34")};
  }
`

const ArtistHeaderSection = styled.div`
  height: 60vh;
  min-height: 450px;
  border: ${BORDER_WIDTH}px solid ${color("purple100")};
  border-left: none;
  display: flex;
  margin-bottom: 40px;
`

const ArtistHeaderTitle = styled.div`
  flex: 1;
  border-right: ${BORDER_WIDTH}px solid ${color("purple100")};

  h1 {
    ${unica("s65")};
    width: 100%;
    height: 50%;
    min-height: fit-content;
    padding: ${space(2)}px;
    border-bottom: ${BORDER_WIDTH}px solid ${color("purple100")};
    &:hover {
      background: ${color("purple100")};
      color: white;
    }
  }

  h2 {
    ${unica("s25")};
    width: 50%;
    height: 50%;
    min-height: fit-content;
    display: inline-flex;
    padding: 20px;
    &:last-child {
      border-left: ${BORDER_WIDTH}px solid ${color("purple100")};
    }
    &:hover {
      background: ${color("purple100")};
      color: white;
    }
  }
`

const ArtistHeaderImg = styled.div<{ src?: string }>`
  flex: 1;
  background: ${color("purple100")};
  ${props =>
    props.src &&
    `
    background: url(${props.src});
    background-size: cover;
    background-position: center;
  `};
`

const IntroSection = styled(Flex)`
  ${BylineContainer} {
    flex-direction: column;
    align-items: flex-start;
  }
`

const TextSection = styled(Serif)`
  max-width: 75%;
  margin-left: auto;
  padding-bottom: 60px;

  p {
    font-size: 24px;
    text-indent: 2em;

    &:first-child {
      text-indent: 0;
    }
  }
`
const ImageSetWrapper = styled.div`
  margin-bottom: 60px;
  border: ${BORDER_WIDTH}px solid ${color("purple100")};
  border-left: none;
`
