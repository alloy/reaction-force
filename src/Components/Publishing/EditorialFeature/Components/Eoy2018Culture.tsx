import {
  ArticleProps,
  FeatureLayout,
} from "Components/Publishing/Layouts/FeatureLayout"
import { Nav } from "Components/Publishing/Nav/Nav"
import React from "react"
import styled from "styled-components"

export const Eoy2018Culture: React.SFC<ArticleProps> = props => {
  return (
    <ArticleWrapper>
      <Nav canFix={false} transparent />
      <FeatureLayout {...props} />
    </ArticleWrapper>
  )
}

const ArticleWrapper = styled.div`
  background: yellow;
  padding-top: 75px;
`
