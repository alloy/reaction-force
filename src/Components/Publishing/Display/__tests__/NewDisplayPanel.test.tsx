import { NewDisplayPanel } from "Components/Publishing/Display/NewDisplayPanel"
import { mount } from "enzyme"
import "jest-styled-components"
import React from "react"
import renderer from "react-test-renderer"

import { StandardArticleHostedAdPanel } from "../../Fixtures/Components"
// tslint:disable: jsx-boolean-value

describe("snapshot", () => {
  it("renders the new canvas in standard layout", () => {
    const component = renderer
      .create(
        <NewDisplayPanel
          adDimension={StandardArticleHostedAdPanel.adDimension}
          adUnit={StandardArticleHostedAdPanel.adUnit}
          displayNewAds={true}
        />
      )
      .toJSON()
    expect(component).toMatchSnapshot()
  })
})

describe("data", () => {
  it("renders the component with the correct data and properties in standard layout articles", () => {
    const panel = mount(
      <NewDisplayPanel
        adDimension={StandardArticleHostedAdPanel.adDimension}
        adUnit={StandardArticleHostedAdPanel.adUnit}
        displayNewAds={true}
      />
    )

    expect(panel.props().adDimension).toEqual("300x250")
    expect(panel.props().adUnit).toEqual("Desktop_RightRail1")
    expect(panel.props().displayNewAds).toBe(true)
    panel.find({ className: "htl-ad" })
    panel.find({ "data-sizes": "300x250" })
    panel.find({ "data-eager": true })
    panel.find({ "data-unit": "Desktop_RightRail1" })
  })
})
