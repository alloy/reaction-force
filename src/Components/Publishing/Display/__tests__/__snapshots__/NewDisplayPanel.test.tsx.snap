// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot renders the new canvas in standard layout 1`] = `
.c1 {
  margin: auto;
}

.c2 {
  font-family: Unica77LLWebRegular,'Helvetica Neue',Helvetica,Arial,sans-serif;
  font-size: 10px;
  line-height: 14px;
  color: black30;
  margin: 4px;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding-bottom: 4px;
  padding-top: 8px;
  background: #F8F8F8;
  text-align: center;
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    <div
      style={
        Object {
          "height": 250,
          "width": 300,
        }
      }
    />
    <div
      className="c2"
      color="black30"
      fontFamily="Unica77LLWebRegular, 'Helvetica Neue', Helvetica, Arial, sans-serif"
      fontSize="10px"
    >
      Advertisement
    </div>
  </div>
</div>
`;
